### DISCUSSION TITLE
nonebot2 钉钉如何主动发起消息

### DISCUSSION THREAD

**Discussion by nanshaobit** [nonebot/discussions#58] [问答] [answered]
**需求**：使用钉钉机器人在定时任务或其他需要主动发起会话的情况下，主动发送消息。
查阅源码得知，目前都是event 驱动，通过`get_bot`也无法获得钉钉bot。

问题：
+ 如何主动发送群聊消息。
+ 以及主动发送单聊接口的实现（ https://open.dingtalk.com/document/robots/chatbots-send-one-on-one-chat-messages-in-batches ）

**Comment by yanyongyu** [comment]
@lengthmin

**Comment by bytemain** [comment, marked as answer]
1. 目前 `get_bot` 还有点问题，预计下个版本修复。
2. 目前可以调用 `bot.send` 向任意 webhook 发送消息，需要传对应的参数即可。见：https://v2.nonebot.dev/guide/ding-guide.html#%E5%8F%91%E9%80%81%E5%88%B0%E7%89%B9%E5%AE%9A%E7%BE%A4%E8%81%8A
3. 单聊接口还未实现。

目前 nonebot2 以及 nonebot-adapter-ding 还处于早期预览版本，部分功能还没有实现。

不过我们接下来会有一波更新，请期待一下。
